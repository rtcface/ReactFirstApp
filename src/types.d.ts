export interface Gifs {
  id    : string
  url   : string
  title : string
}

export interface LoadGifts {
    images:Gifs
    isLoading:boolean
}


export interface Gifts {
  id    : string
  url   : string
  title : string
  images: images  
}

interface images {
    downsized_medium : downsized_medium
}

interface downsized_medium{
    url:string
}


