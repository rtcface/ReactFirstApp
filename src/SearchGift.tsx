import { useState, useEffect } from 'react';    
import { GifGrid,AddCategory }  from './compounents';

interface SearchGif{
  categories: Array<string>
}

function SearchGift():JSX.Element {
  const INITIAL_STATE = ['Avengers','Movies'];
  const [categories, setCategories] = useState<SearchGif["categories"]>([]);

  useEffect(() => {
    setCategories(INITIAL_STATE);
  },[]);
  
const addCategorie = ( newCategory:string ) => {
  if ( categories.includes(newCategory) ) return;
  setCategories( [newCategory,...categories] );
  
};


  return (
    <>
      <h1>Buscador de gifs</h1>

      <AddCategory onNewCategories={ ( categorie:string ) => addCategorie(categorie) } />      
        {
          categories.map( item => <GifGrid key={item} categorie={item}/> ) 
      }

    </>
  );
}


export default SearchGift;
