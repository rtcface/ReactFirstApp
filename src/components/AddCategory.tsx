import {useState} from 'react';

export interface AddCategory {
categoryName:string
}

export interface AddCategoryProps{
  onNewCategories: ( categoryName:string ) => void
}



export function AddCategory({ onNewCategories }:AddCategoryProps) {

  const [inputValue, setInputValue] = useState<AddCategory["categoryName"]>('');

const onChangeCat = ({target}:React.ChangeEvent<HTMLInputElement>) => {
  setInputValue( target.value ); 
  //console.log('data', inputValue)
};

const onSubmitCat = ( event:React.FormEvent<HTMLFormElement> ) => {
  event.preventDefault();
  //console.log('input',inputValue);
  
  if( inputValue.trim().length <= 3 ) return;

  onNewCategories( inputValue.trim()  );

  setInputValue('');

};

  return (
      <form onSubmit={ onSubmitCat }>
        <input 
          type='text' 
          placeholder='Escribe la categoria'
          value={ inputValue }
          onChange={ onChangeCat  }
        />
      </form>
  );
}
