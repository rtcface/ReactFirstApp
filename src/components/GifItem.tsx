 
import { Gifs } from "../types";

export function GifItem({ title,url }:Gifs):JSX.Element {
	return (
	<div className='card'>
		<img src={ url } alt={ title } />	
		<p>{title}</p>
		</div>
	);
}

