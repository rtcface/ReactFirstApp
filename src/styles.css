* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}

.card p {
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}


.lds-facebook {
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
}
.lds-facebook div {
  display: inline-block;
  position: absolute;
  left: 8px;
  width: 16px;
  background: #fff;
  animation: lds-facebook 1.2s cubic-bezier(0, 0.5, 0.5, 1) infinite;
}
.lds-facebook div:nth-child(1) {
  left: 8px;
  animation-delay: -0.24s;
}
.lds-facebook div:nth-child(2) {
  left: 32px;
  animation-delay: -0.12s;
}
.lds-facebook div:nth-child(3) {
  left: 56px;
  animation-delay: 0;
}
@keyframes lds-facebook {
  0% {
    top: 8px;
    height: 64px;
  }
  50%, 100% {
    top: 24px;
    height: 32px;
  }
}
