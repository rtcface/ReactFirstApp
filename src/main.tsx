import React from 'react'
import ReactDOM from 'react-dom/client'
import SearchGift from './SearchGift'
import './styles.css'

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    <SearchGift  />
  </React.StrictMode>
)
