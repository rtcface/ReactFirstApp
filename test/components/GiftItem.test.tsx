import { render } from "@testing-library/react";
import React from "react";
import { GifItem } from '../../src/components/GifItem';
import { Gifs } from "../../src/types";

describe('test of camponent GiftItem', () => { 
    
    test('should to match with the snapshot', () => {
        const defaultProps: Gifs = {
            id:"2",
            title:"Avatars",
            url:"https://giphy.com/gifs/TOEIAnimationUK-goku-dragon-ball-super-broly-WOb8EeFziTQNE02WXs"
        };
        const { container } = render(<GifItem { ...defaultProps }/>);
        expect( container ).toMatchSnapshot();
        
    });
     
});


