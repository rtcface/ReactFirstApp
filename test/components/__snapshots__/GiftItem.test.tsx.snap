// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of camponent GiftItem should to match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Avatars"
      src="https://giphy.com/gifs/TOEIAnimationUK-goku-dragon-ball-super-broly-WOb8EeFziTQNE02WXs"
    />
    <p>
      Avatars
    </p>
  </div>
</div>
`;
